hal/audio_extn/cirrus_sony.c

```diff
@@ -600,35 +600,64 @@ static int cirrus_play_silence(int seconds) {
     struct audio_device *adev = handle.adev_handle;
     struct mixer_ctl *ctl_config = NULL;
     struct pcm_config rx_tmp = { 0 };
+    struct audio_usecase *uc_info_rx;
 
     uint8_t *silence = NULL;
-    int i, ret, silence_bytes, silence_cnt = 1;
+    int i, ret = 0, silence_bytes, silence_cnt = 1;
     unsigned int buffer_size = 0, frames_bytes = 0;
+    int pcm_dev_rx_id, adev_retry = 5;
 
+    if (!list_empty(&adev->usecase_list)) {
+        ALOGD("%s: Usecase present retry speaker protection", __func__);
+        return -EAGAIN;
+    }
 
-    ret = cirrus_set_mixer_value_by_name(CIRRUS_PLAYBACK_MIXER, 1);
-    if (ret)
-        return ret;
+    uc_info_rx = (struct audio_usecase *)calloc(1, sizeof(struct audio_usecase));
+    if (!uc_info_rx) {
+        return -ENOMEM;
+    }
+
+    while ((!adev->primary_output || !adev->platform) && adev_retry) {
+        ALOGI("%s: Waiting for audio device...", __func__);
+        sleep(1);
+        adev_retry--;
+    }
 
-    handle.pcm_rx = pcm_open(adev->snd_card, 0,
-                             PCM_OUT, &pcm_config_cirrus_rx);
+    uc_info_rx->id = USECASE_AUDIO_PLAYBACK_DEEP_BUFFER;
+    uc_info_rx->type = PCM_PLAYBACK;
+    uc_info_rx->in_snd_device = SND_DEVICE_NONE;
+    uc_info_rx->stream.out = adev->primary_output;
+    list_init(&uc_info_rx->device_list);
+    uc_info_rx->out_snd_device = SND_DEVICE_OUT_SPEAKER_PROTECTED;
```

Comment:
@kholk I've changed this to protected again. Normal speaker doesn't work, and we've seen before that the HAL maps speaker to speaker_protected in _most_ scenarios _anyway_ because we're a "spkr prot".